packages/adblocker-extended-selectors/src/eval.ts

```diff
@@ -8,6 +8,49 @@
 
 import type { AST, Complex } from './types.js';
 
+/**
+ * Evaluates an XPath expression and returns matching Element nodes.
+ * @param element - The context element for XPath evaluation
+ * @param xpathExpression - The XPath expression to evaluate
+ * @returns Array of Element nodes that match the XPath expression
+ */
+function handleXPathSelector(element: Element, xpathExpression: string | undefined): Element[] {
+  if (xpathExpression === undefined) {
+    return [];
+  }
+
+  try {
+    const document = element.ownerDocument;
+    if (!Node || !XPathResult || !document || typeof document.evaluate !== 'function') {
```

Comment:
I assume checks like "!Node" and "!XpathResult" are feature detection, but it is not intuitive why they are needed. Since this is very platform specific, can you add comments what assumption you want to check here.

Also, since "XPathResult.ORDERED_NODE_SNAPSHOT_TYPE" is then used below, does that need to be detected as well by these checks?